### PR TITLE
Add HCL2 aws_secretsmanager function

### PR DESCRIPTION
**Pull request by azr**
Implements aws_secretsmanager function for HCL. Adds docs.

Checks a box on #9176

### CODE REVIEW

**Comment by azr** [comment]
( still have to test this 🙂  )

**Review by nywilken — COMMENTED** [review]
This is looking very good. There is a template that you can test against in our test templates repo. My only comment so far is that the function `aws_secretsmanager` is not registered in hcl2template/functions.

**Comment by azr** [comment]
Good catch thanks !

File: website/pages/docs/from-1.5/functions/contextual/aws_secretsmanager.mdx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+sidebar_title: aws_secretsmanager
+description: The aws_secretsmanager function retrieves secrets from Amazon secretsmanager stores.
+---
+
+
+# `aws_secretsmanager_key` Function
+
+Secrets can be read from the [AWS Secrets
+Manager](https://aws.amazon.com/secrets-manager/) and used within your template
+as variables.
```
Comment (nywilken) [inline website/pages/docs/from-1.5/functions/contextual/aws_secretsmanager.mdx:13]:
```suggestion
as locals.
```

When using the hcl2_upgrade command to convert the template the function is placed in a variable block which is expected. But was a little confusing to me at first. Your example below is perfect so I would just suggest changing this to `locals`

**Review by nywilken — APPROVED** [review]
Nicely done. Good catch on the panic. One small suggestion but otherwise good to go.

```
⇶  packer.test build amazon-ebs_secretsmanager_shell-local.json.pkr.hcl
null.autogenerated_1: output will be in this color.

==> null.autogenerated_1: Running local shell script: /tmp/packer-shell629348826
    null.autogenerated_1: boo
    null.autogenerated_1: shell:powershell
    null.autogenerated_1: test:kitchen
Build 'null.autogenerated_1' finished after 283 milliseconds 706 microseconds.
```

File: website/pages/docs/from-1.5/functions/contextual/aws_secretsmanager.mdx (hunk 1 above)
Comment (azr) [reply website/pages/docs/from-1.5/functions/contextual/aws_secretsmanager.mdx:13]: Good call totally, variables cannot be made of functions

File: template/interpolate/aws/secretsmanager/secretsmanager.go [hunk 2]
```diff
@@ -53,6 +53,9 @@ func (c *Client) GetSecret(spec *SecretSpec) (string, error) {
 		SecretId:     aws.String(spec.Name),
 		VersionStage: aws.String("AWSCURRENT"),
 	}
+	if spec.Name != "" {
+		params.VersionStage = aws.String(spec.Key)
+	}
```
Comment (azr) [inline template/interpolate/aws/secretsmanager/secretsmanager.go:58]: Just FYI, The version parameter was ignored before !

**Comment by azr** [comment]
Thanks for reviewing @nywilken 🙂

**Pull request merged**